### PR TITLE
Implement integration tests for authn method registration through session.

### PR DESCRIPTION
**Pull request by sea-snake**
Implement integration tests for authn method registration through session.

# Changes

- Added optional `authn_method` param to `api_v2::authn_method_registration_mode_exit`.
- Added `authn_method_session_register` to `api::v2`.
- Added tests in `authn_method_session_registration.rs` to cover new session cases not covered in `authn_method_registration.rs`.

### CODE REVIEW

File: src/internet_identity/tests/integration/v2_api/authn_method_session_registration.rs
Comment (lmuntaner) [inline src/internet_identity/tests/integration/v2_api/authn_method_session_registration.rs:1]: How about adding the following tests:

* Two different users can add a device at the same time.
* The code from one user doesn't work with another user.
* Exiting mode in one user doesn't exist in the other.

Thanks!

Comment (sea-snake) [reply src/internet_identity/tests/integration/v2_api/authn_method_session_registration.rs:1]: These should already be covered in the other e2e tests file. But I'll add them given that sessions are involved and could have edge cases. Thanks!

Comment (lmuntaner) [reply src/internet_identity/tests/integration/v2_api/authn_method_session_registration.rs:1]: Ah, yeah, I guess just copy paste them. It should be easy to have them...

Comment (sea-snake) [reply src/internet_identity/tests/integration/v2_api/authn_method_session_registration.rs:1]: > Two different users can add a device at the same time.

Added `should_register_authn_methods_in_parallel` in `authn_method_session_registration.rs`

> The code from one user doesn't work with another user.

Added `should_reject_code_from_another_registration` in `authn_method_session_registration.rs`

> Exiting mode in one user doesn't exist in the other.

Added `should_exit_registrations_separately` in `authn_method_session_registration.rs`

File: src/internet_identity/tests/integration/v2_api/authn_method_session_registration.rs [hunk 1]
```diff
@@ -0,0 +1,617 @@
...
+    )?
+    .expect("authn_method_registration_mode_exit failed for identity 1");
+    api_v2::authn_method_confirm(
+        &env,
+        canister_id,
+        authn_method2.principal(),
+        identity_number2,
+        &add_response1.confirmation_code,
+    )?
+    .expect("authn_method_confirm failed for identity 2");
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply src/internet_identity/tests/integration/v2_api/authn_method_session_registration.rs]: This seems right, it should fail as is now, right?

**Review by lmuntaner — COMMENTED** [review]
One comment on the bot's comment as well

File: src/internet_identity/tests/integration/v2_api/authn_method_session_registration.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (sea-snake) [reply src/internet_identity/tests/integration/v2_api/authn_method_session_registration.rs]: Yes, good catch by the bot.

**Review by lmuntaner — APPROVED** [review]
